### PR TITLE
Cap DuckDB threads and memory per connection

### PR DESCRIPTION
**Pull request by jcelliott**
DuckDB sized each connection's thread pool and memory limit to the host, and we cache up to 100 connections. On a 14-core host that measured 1040 parked scheduler threads, and the ceiling grew with the core count rather than the workload.

Both are now set explicitly: 4 worker threads and 512MiB per connection. Dataframes below DuckDB's 122,880-row row group are scanned single-threaded regardless, so the surplus workers earned nothing at the sizes we index; bulk indexing keeps meaningful parallelism. The memory limit is a ceiling, not a reservation, and these databases are file-backed, so a query over it spills to disk rather than failing.

Also fixes an unrelated race in the flush test, which asserted the process-wide connection cache was empty when other tests running in parallel could repopulate it.

Verified by a test covering both settings on both connection kinds, and by running the server for 17 minutes under load with no regression (threads peaked at 77, against 876 on main, though the workloads differed so this is not a controlled comparison).

### CODE REVIEW

File: crates/liboxen/src/core/db/data_frames/df_db.rs [hunk 1]
```diff
@@ -1222,6 +1250,60 @@ mod tests {
 
     use super::*;
 
+    /// DuckDB reports `memory_limit` as a formatted binary size, so match on the
+    /// value and unit rather than on its exact rendering.
+    fn assert_memory_capped(reported: &str, kind: &str) {
+        assert!(
+            reported.contains("512") && reported.contains("MiB"),
+            "{kind} connections must cap memory at {DUCKDB_MAX_MEMORY}, got {reported:?}"
```
[the remark this answers is not in the document]
Comment (jcelliott) [reply crates/liboxen/src/core/db/data_frames/df_db.rs]: Fixed in ca30a135c. `contains("512")` did accept `"5120.0 MiB"`, so a connection 10x over the cap would have passed. It now parses both the configured and reported sizes into magnitude and unit and compares those, which stays tolerant of DuckDB's formatting without widening what it accepts. Confirmed non-vacuous by dropping the cap locally: the test fails with `got "28.7 GiB"`.

File: crates/liboxen/src/core/db/data_frames/df_db.rs [hunk 2]
```diff
@@ -1222,6 +1250,60 @@ mod tests {
...
+            let conn = get_connection(&db_file)?;
+            conn.execute_batch("CREATE TABLE df AS SELECT 42 AS x")?;
+            let rw_threads: i64 = conn.query_row(
+                "SELECT CAST(current_setting('threads') AS BIGINT)",
+                [],
+                |row| row.get(0),
+            )?;
+            assert_eq!(
+                rw_threads, DUCKDB_THREADS,
+                "read-write connections must cap the thread pool"
```
[the remark this answers is not in the document]
Comment (jcelliott) [reply crates/liboxen/src/core/db/data_frames/df_db.rs:1260]: Declining this one. `DUCKDB_THREADS` is a tuning constant, not a contract fixed at five, so a literal would fail the test on any deliberate retune without catching a defect. What the test is for is that the configured value reaches both connection kinds and survives `lock_configuration` on the hardened one, which is the part that could regress invisibly.

**Review by CleanCut — APPROVED** [review]
